packages/nx/src/command-line/release/utils/shared.ts

```diff
@@ -229,6 +234,33 @@ function stripPlaceholders(str: string, placeholders: string[]): string {
   return str;
 }
 
+export function shouldPreferDockerVersionForReleaseGroup(
+  releaseGroup: ReleaseGroupWithName
+): boolean {
+  // Check if the release group has docker configuration at all
+  if (!releaseGroup.docker) {
+    // No docker configuration means we should use the inferred setting
+    // The inference was already done in the config phase, so if docker projects exist,
+    // releaseTagPatternRequireSemver would be false
+    return !releaseGroup.releaseTagPatternRequireSemver;
+  }
+
+  // The inference was already done in the config phase, so if docker projects exist,
+  // releaseTagPatternRequireSemver would be false
+  return !releaseGroup.releaseTagPatternRequireSemver;
+}
+
+export function shouldSkipVersionActions(
+  dockerOptions: { skipVersionActions?: string[] | boolean },
+  projectName: string
+): boolean {
+  return (
+    dockerOptions.skipVersionActions === true ||
+    (Array.isArray(dockerOptions.skipVersionActions) &&
+      dockerOptions.skipVersionActions.includes(projectName))
```

Comment:
Let's use proper project matches here not just exact strings, so that we have the same flexibility as in `projects`

Reply:
I'm not sure where we are doing this for `projects` in nx release currently. It seems like we're matching by full name for `args.projects` too?

Reply:
Found it